foundation/foundation-mda/src/main/resources/templates/data-delivery-pyspark/asynchronous.processor.base.py.vm

```diff
@@ -295,7 +295,7 @@ class ${step.capitalizedName}Base(AbstractPipelineStep):
             job_name = self.get_job_name()
             default_namespace = self.get_default_namespace()
             parent_run_facet = PipelineBase().get_pipeline_run_as_parent_run_facet()
-            event_data = self.create_base_lineage_event_data()
+            event_data = self.create_base_lineage_event_data() if self.create_base_lineage_event_data() else None
```

Comment:
Q: What's the goal of this change? Looks like we just end up redundantly calling `create_...data`

Reply:
This was flagged by Pylint because the function `create_base_lineage_event_data` by returns a None. The suggested changes reflects Pylint's suggested [documentation for assignment-from-none](https://pylint.readthedocs.io/en/stable/user_guide/messages/error/assignment-from-none.html). Personally (and I think some folks on SO) mentioned this feels like a Refactor rather than an Error, but Pylint has it as an Error. Another workaround is to disable the assignment-from-none for that code.

Reply:
Do you know if this is special assignment syntax for Python that avoids double-calling the function, or is this actually going to:
1. call the function
2. see it returned a non-null value
3. call the function again and use that in the assignment

If it's doing the latter, I think we probably want to just disable the check here.  Because we _expect_ that this method will most often be overloaded and provide real logic other than `return None`.  We could also return some empty-but-not-null value in the default case.

Reply:
I did a little testing with timing and the timing results indicate that the function get's triggered twice since the execution time was doubled. I have made the changes to disable the checker.